src/actions/accountsActions.js

```diff
@@ -269,23 +300,13 @@ export const initOnLoginSmartWalletAccountAction = (privateKey: string) => {
       accounts: {
         data: accounts,
       },
-      smartWallet: {
-        upgrade: {
-          status: upgradeStatus,
-        },
-      },
     } = getState();
 
     const activeAccountId = getActiveAccountId(accounts);
     const activeAccountType = getActiveAccountType(accounts);
 
     if (activeAccountType !== ACCOUNT_TYPES.SMART_WALLET) {
-      if ([
-        SMART_WALLET_UPGRADE_STATUSES.TRANSFERRING_ASSETS,
-        SMART_WALLET_UPGRADE_STATUSES.DEPLOYING,
-      ].includes(upgradeStatus)) {
-        await dispatch(initSmartWalletSdkAction(privateKey));
-      }
+      await dispatch(initSmartWalletSdkAction(privateKey));
```

Comment:
I guess this will now force app to switch to Smart Wallet even if user left the app in legacy wallet?

Reply:
It does not. This method only initialises sdk and sets it as init on redux store.
After doing so we return (so connectSmartWalletAccountAction is not called if previously used wallet wasn't smart wallet).